src/wallet/wallet_tx_builder.cpp

```diff
@@ -55,15 +55,20 @@ WalletTxBuilder::GetChangeAddress(
         return result;
     };
 
-    auto changeAddressForTransparentSelector = [&](const std::set<ReceiverType>& receiverTypes) {
+    auto changeAddressForTransparentSelector = [&](const std::set<ReceiverType>& receiverTypes)
+        -> tl::expected<ChangeAddress, AddressResolutionError> {
         auto addr = wallet.GenerateChangeAddressForAccount(
                 sendFromAccount,
                 getAllowedChangePools(receiverTypes));
-        assert(addr.has_value());
-        return addr.value();
+        if (addr.has_value()) {
+            return {addr.value()};
+        } else {
+            return tl::make_unexpected(AddressResolutionError::TransparentChangeNotAllowed);
+        }
     };
 
-    auto changeAddressForSaplingAddress = [&](const libzcash::SaplingPaymentAddress& addr) -> RecipientAddress {
+    auto changeAddressForSaplingAddress = [&](const libzcash::SaplingPaymentAddress& addr)
+        -> tl::expected<ChangeAddress, AddressResolutionError> {
```

Comment:
This cannot return an `unexpected` so its return type should remain `RecipientAddress` (or if not then the body is wrong).